hudi-client/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackHelper.java

```diff
@@ -130,39 +137,55 @@ public ListingBasedRollbackHelper(HoodieTableMetaClient metaClient, HoodieWriteC
               1L
           );
           return new Tuple2<>(rollbackRequest.getPartitionPath(),
-                  HoodieRollbackStat.newBuilder().withPartitionPath(rollbackRequest.getPartitionPath())
-                          .withRollbackBlockAppendResults(filesToNumBlocksRollback).build());
+              HoodieRollbackStat.newBuilder().withPartitionPath(rollbackRequest.getPartitionPath())
+                  .withRollbackBlockAppendResults(filesToNumBlocksRollback).build());
         }
         default:
           throw new IllegalStateException("Unknown Rollback action " + rollbackRequest);
       }
-    }).reduceByKey(RollbackUtils::mergeRollbackStat).map(Tuple2::_2).collect();
+    });
   }
 
 
-
   /**
    * Common method used for cleaning out base files under a partition path during rollback of a set of commits.
    */
-  private Map<FileStatus, Boolean> deleteCleanedFiles(HoodieTableMetaClient metaClient, HoodieWriteConfig config,
-                                                      String partitionPath, PathFilter filter) throws IOException {
+  private Map<FileStatus, Boolean> deleteBaseAndLogFiles(HoodieTableMetaClient metaClient, HoodieWriteConfig config,
```

Comment:
can you help me understand how to differentiate between CREATE and MERGE in these code blocks.

Reply:
its hard. but would nt MERGE work for both in terms of actually performing a correct rollback?